.github/workflows/preview-release.yml

```
@@ -74,28 +74,13 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ steps.generate_github_token.outputs.token }}
 
-      - name: Create Release Pull Request or Publish to npm
-        id: changesets
-        uses: dotansimha/changesets-action@v1.5.2
-        with:
-          commit: 'ci(changesets): version packages'
-          publish: yarn changeset publish
-          version: yarn changeset:version-and-format
-          createGithubReleases: aggregate
-          githubReleaseName: v${{ steps.release_version.outputs.VALUE }}
-          githubTagName: v${{ steps.release_version.outputs.VALUE }}
-        env:
-          GITHUB_TOKEN: ${{ steps.generate_github_token.outputs.token }}
-          SKIP_POSTINSTALL_DEV_SETUP: true
-
-      # Publish canary releases only if the packages weren't published already
+      # Publish preview releases only if the branch starts with "preview/" and packages weren't published already
       - name: Publishing preview releases to npm registry
-        if: steps.changesets.outputs.published != 'true' && startsWith(github.ref, 'refs/heads/preview/')
+        if: startsWith(github.ref, 'refs/heads/preview/')
```

what if we make publishing opt-in instead of publishing every commit to branches with open PRs?

That would drastically reduce the amount of artifacts we are publishing to npm, reduce the amount of compute used per action, and still allow for preview publishing whenever needed.
```suggestion
        if: startsWith(github.ref, 'refs/heads/preview/')
        if: contains(github.event.head_commit.message, '[publish_preview]'
```

---

Thats a good idea. So if we want to trigger a version then we add a trigger string to the commit message too. Makes it more granular.
Thank you.

---

updated here: https://github.com/commercetools/ui-kit/pull/2916/commits/ac083e1f7f7bf4dd1996adc851748c31a7eaa3bc